aocd/models.py

```diff
@@ -39,16 +55,28 @@
 AOCD_CONFIG_DIR = Path(os.environ.get("AOCD_CONFIG_DIR", AOCD_DATA_DIR)).expanduser()
 URL = "https://adventofcode.com/{year}/day/{day}"
 
+_TUser = TypeVar("_TUser", bound="User") # TODO: Use typing.Self instead of this when support for < 3.11 is dropped
```

Comment:
We could also use `typing_extensions.Self` immediately, but I don't think it makes much of a difference and I figured it'd be better to avoid the dependency (at least, on recent Python versions).

Reply:
It looks like you already have the `'typing-extensions; python_version < "3.10"',` dependency in pyproject.toml, so go ahead and use it.